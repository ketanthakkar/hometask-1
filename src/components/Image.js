import React from 'react';
import '../styleModules/Image.css';

const Image = ({src, alt}) => {
    return <img src = {src} alt = {alt} className = 'image__small'/>
}

export default Image;
