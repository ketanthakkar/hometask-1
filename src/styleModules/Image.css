.image__small {
    width:200px;
    height: 300px;
}