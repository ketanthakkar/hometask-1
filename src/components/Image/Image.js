import React from 'react';
import './Image.css';

const Image = ({src, alt}) => {
    return <img src = {src} alt = {alt} className = 'image__small'/>
}

export default Image;
