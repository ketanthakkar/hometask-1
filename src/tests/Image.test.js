import React from 'react';
import { shallow } from 'enzyme';
import Image from '../components/Image';

describe('Image', () => {
  it('shoud render image component', () => {
    const render = shallow(
      <Image src = 'src' alt = 'alt' />
    );
    expect(render.find('img').hasClass('image__small')).toBe(true);
    expect(render).toMatchSnapshot();
  })

  it('should not modify props', () => {
    const src = 'src', alt = 'alt';
    const render = shallow(
      <Image src = {src} alt = {alt} />
    );
    expect(src).toEqual('src');
    expect(alt).toEqual('alt');
    expect(render).toMatchSnapshot();
  })
})