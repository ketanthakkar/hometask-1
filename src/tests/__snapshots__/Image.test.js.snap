// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image shoud render image component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    alt="alt"
    src="src"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "alt",
      "className": "image__small",
      "src": "src",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "alt",
        "className": "image__small",
        "src": "src",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Image should not modify props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    alt="alt"
    src="src"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "alt",
      "className": "image__small",
      "src": "src",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "alt",
        "className": "image__small",
        "src": "src",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
